### DISCUSSION TITLE
Solve two field variables with separator solvers

### DISCUSSION THREAD

**Discussion by avtarsinghh1991** [idaholab/moose#20069] [Q&A Modules: General] [answered]
Hello MOOSE Experts,

I am working on a multi-field problem in which one field converges with a different solver or preconditioner and another one with a different solver or preconditioner. Is there any way by which I can solve different fields with separate solvers?

I am thinking of using the MultiApp module provided in the MOOSE.
Please let me know. I shall be very thankful.

Best
Avtar

**Comment by GiudGiud** [comment, marked as answer]
Hello

Multiapps are the current cannon way to do this. You can converge all fields by iterating between both apps (tight coupling).
This tutorial should help: https://mooseframework.inl.gov/getting_started/examples_and_tutorials/tutorial02_multiapps/index.html

If you think you can get away with different pre-conditioners for each field, while keeping them fully-coupled (single matrix solve), then you may use a field-split preconditioning instead. I wrote some docs but they arent merged yet. They are online on this PR:
https://github.com/idaholab/moose/blob/585373f878996b8d56f3b96bb6fd04ad4c8b036f/framework/doc/content/source/preconditioners/FieldSplitPreconditioner.md
Doc website isnt up yet

Guillaume

**Reply by avtarsinghh1991** [reply]
Hello Guillaume

Thank you very much for guiding me.

Best
Avtar
